### ISSUE TITLE
Port changes from CSoar revision 12269 (total order on symbols)

### ISSUE THREAD

**Issue by voigtjr** [soartech/jsoar#30]
https://code.google.com/p/soar/source/detail?r=12269

commit message: "inequality comparisons on string constants and identifiers are now evaluated lexically rather than always returning false"

Comment 3 by project member @marinier, Jun 2, 2011

Looks like the place where comparisons actually takes place in jsoar is different. SymbolImpl contains methods like numericGreater, numericLess, etc., which all return false. Then the specific numeric symbol subclasses (DoubleSymbolImpl and IntegerSymbolImpl), override these to do the actual comparison, while everyone else defaults to false.

So basically, the single function in csoar probably belongs in SymbolImpl, and then the overrides from the subclasses can be removed.

My plan is to create a private function that performs the comparison, and then to have each of these comparison methods call it.

**Comment by marinier** [comment]
Total ordering on identifiers has been added (commit 7bf2d0f6). Not yet pushed publicly.

**Comment by marinier** [comment]
Total ordering on strings has been added (commit 8b893c7). Not yet pushed publicly, but otherwise this should now be complete.

FYI, my approach was just to override in the subclasses as per the original design, not to move the logic to SymbolImpl as originally stated back in 2011.
